### PR TITLE
feat: Add test to protocol contracts

### PR DESCRIPTION
**Pull request by andresaiello**
### Summary
- Add test to protocol contracts
- Fix some validations already discussed with the protocol team

Test added:
  ConnectorZEVM tests
    ZetaConnectorZEVM
      ✔ Should revert if the zetaTxSender has no enough zeta
      ✔ Should revert if the zetaTxSender didn't allow ZetaConnector to spend Zeta token
      ✔ Should emit `ZetaSent` on success
      ✔ Should transfer value and gas to fungible address
      ✔ Should update wzeta address if is call from fungible address
      ✔ Should revert if try to update wzeta address from other address

  ERC20Custody tests
    ERC20Custody
      ✔ Should update the tss address
      ✔ Should revert if updateTSSAddress not called by tssUpdater
      ✔ Should update zeta fee
      ✔ Should revert if updateZetaFee not called by tssUpdater
      ✔ Should revert if updateZetaFee is higher than max
      ✔ Should renounce tssUpdater
      ✔ Should whitelist
      ✔ Should revert if whitelist is not called by tss
      ✔ Should unwhitelist
      ✔ Should revert if unwhitelist is not called by tss
      ✔ Should emit Deposit event on deposit
      ✔ Should emit Deposit event and do the right math if is called twice (43ms)
      ✔ Should revert deposit if is paused
      ✔ Should revert deposit if is not whitelisted
      ✔ Should revert deposit if has no zeta to pay fee
      ✔ Should emit Withdrawn event
      ✔ Should revert if withdraw is not called by tss
      ✔ Should not revert withdraw if is paused
      ✔ Should revert withdraw if is not whitelisted
      ✔ Should not allow reentrant calls to Deposit method
      ✔ Should not allow reentrant calls to Witdraw method

  ZRC20 tests
    ZRC20Contract
      ✔ Should return name
      ✔ Should return symbol
      ✔ Should return decimals
      ✔ Should return chainId
      ✔ Should return coinType
      ✔ Should return gasLimit
      ✔ Should return systemContractAddress
      ✔ Should return totalSupply
      ✔ Should return balanceOf
      ✔ Should return allowance
      ✔ Should deposit if is called by fungible module
      ✔ Should not deposit if is not called by fungible module
      ✔ Should transfer

  40 passing (6s)

### CODE REVIEW

File: contracts/zevm/testing/MockSystemContract.sol [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+    error CallerIsNotFungibleModule();
+
+    error InvalidTarget();
+
+    error CantBeIdenticalAddresses();
+
+    error CantBeZeroAddress();
+}
+
+contract MockSystemContract is SystemContractErrors {
```
Comment (lumtis) [inline contracts/zevm/testing/MockSystemContract.sol:17]: Maybe we can add `is zContract`?

**Review by lumtis — APPROVED** [review]
Looks good to me.
I think we could add test as part of the CI in `.github`

File: contracts/zevm/testing/MockSystemContract.sol (hunk 1 above)
Comment (andresaiello) [reply contracts/zevm/testing/MockSystemContract.sol:17]: but is not a zContract. maybe you mean `is SystemContract`, but the problem with that is I may change a lot of stuff on SystemContract to make it virtual. I'm trying to polish as much as I can with minimum changes

**Review by lucas-janon — APPROVED** [review]
LGTM

**Review by fadeev — APPROVED** [review]

**Pull request merged**